nucleus/annotation.py

```diff
@@ -62,35 +65,25 @@ def to_payload(self) -> dict:
         return payload
 
 
+@dataclass
 class SegmentationAnnotation(Annotation):
-    def __init__(
-        self,
-        mask_url: str,
-        annotations: List[Segment],
-        reference_id: Optional[str] = None,
-        item_id: Optional[str] = None,
-        annotation_id: Optional[str] = None,
-    ):
-        super().__init__()
-        if not mask_url:
+    mask_url: str
+    annotations: List[Segment]
+    reference_id: Optional[str] = None
+    item_id: Optional[str] = None
+    annotation_id: Optional[str] = None
+
+    def __post_init__(self):
+        if not self.mask_url:
             raise Exception("You must specify a mask_url.")
-        if bool(reference_id) == bool(item_id):
```

Comment:
Is this deletion purposeful? I think UX hypothesis behind this check is that it's much better user experience to know your SegmentationAnnotation is malformed before sending the API request.